src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,317 @@
+/* eslint-disable @typescript-eslint/prefer-optional-chain */
+import { useEffect, useState } from 'react';
+import { Grid, Stack, Box, Typography, Container, Link } from '@mui/material';
+import {
+  type CompanyLinkType,
+  type PreLoginFooterLinksType,
+  type FooterLinksType,
+} from './index';
+import { LangSwitch, type LangSwitchProps } from './LangSwitch';
+import { isRight, toError } from 'fp-ts/lib/Either';
+import { hrefNoOp, wrapHandleExitAction } from '../../utils/index';
+
+/* Icons */
+import TwitterIcon from '@mui/icons-material/Twitter';
+import LinkedIcon from '@mui/icons-material/LinkedIn';
+import InstagramIcon from '@mui/icons-material/Instagram';
+
+/* Enum */
+import { ProductArrayType } from './ProductType';
+import { FundedByNextGenerationEU } from '../../assets/FundedByNextGenerationEU';
+import { LogoPagoPACompany } from '../../assets/LogoPagoPACompany';
+
+type FooterProps = LangSwitchProps & {
+  companyLink: CompanyLinkType;
+  legalInfo: JSX.Element | JSX.Element[];
```

Comment:
Replace JSX.Element with the Generic type defined inside types/components

Reply:
done